src/core/feature/LOD.ts

```
@@ -250,99 +226,57 @@ InstancedMesh2.prototype.addLevel = function (renderList: LODRenderList, geometr
   return object;
 };
 
-InstancedMesh2.prototype.changeLevel = function (
-  renderList,
-  levelIndex,
-  distance,
-  hysteresis
-) {
-  if (!renderList?.levels) throw new Error('Invalid LOD list.');
-  const levels = renderList.levels;
-  if (levelIndex < 0 || levelIndex >= levels.length)
-    throw new Error('Level index OOB');
+InstancedMesh2.prototype.updateLevel = function (renderList, levelIndex, distance, hysteresis, sort = true) {
+  if (!renderList) throw new Error('Render list is invalid.');
 
-  const entry = levels[levelIndex];
+  const level = renderList.levels[levelIndex];
+  if (!level) throw new Error('Cannot update an empty LOD.');
 
   if (distance != null) {
-    const d2 = distance ** 2;
-    if (Number.isNaN(d2)) throw new Error('Distance is NaN');
-    entry.distance = d2;
+    level.distance = distance ** 2; // to avoid to use Math.sqrt every time
+  }
+
+  if (hysteresis != null) {
+    level.hysteresis = hysteresis;
   }
-  if (hysteresis != null) entry.hysteresis = hysteresis;
+
+  if (sort) {
+    renderList.levels.sort((a, b) => a.distance - b.distance);
```

Why sort? 
This method doesn’t change geometry, it only changes distance and hysteresis. If the user gives a bigger number to the next LOD level, we can throw an error; sorting will break the whole LOD level geometry ordering.